src/Compilers/VisualBasic/Portable/Generated/VisualBasic.Grammar.g4

```diff
@@ -0,0 +1,1461 @@
+  ;
+
+option_statement
+  : 'Option' ('Explicit' | 'Strict' | 'Compare' | 'Infer') ('On' | 'Off' | 'Text' | 'Binary')?
+  ;
+
+imports_statement
+  : 'Imports' (imports_clause(',' imports_clause)*)?
+  ;
+
+imports_clause
+  : simple_imports_clause
+  | xml_namespace_imports_clause
+  ;
+
+simple_imports_clause
+  : import_alias_clause? name
+  ;
+
+import_alias_clause
+  : identifier_token '='
+  ;
+
+xml_namespace_imports_clause
+  : '<' xml_attribute '>'
+  ;
+
+xml_attribute
+  : xml_node '=' xml_node
+  ;
+
+attributes_statement
+  : attribute_list*
+  ;
+
+attribute_list
+  : '<' (attribute(',' attribute)*)? '>'
+  ;
+
+attribute
+  : attribute_target? type argument_list?
+  ;
+
+attribute_target
+  : 'Assembly' ':'
+  | 'Module' ':'
+  ;
+
+argument_list
+  : '(' (argument(',' argument)*)? ')'
+  ;
+
+argument
+  : omitted_argument
+  | range_argument
+  | simple_argument
+  ;
+
+omitted_argument
+  : punctuation
+  ;
+
+range_argument
+  : expression 'To' expression
+  ;
+
+simple_argument
+  : name_colon_equals? expression
+  ;
+
+name_colon_equals
+  : identifier_name ':='
+  ;
+
+identifier_name
+  : identifier_token
+  ;
+
+statement
+  : case_statement
+  | catch_statement
+  | declaration_statement
+  | do_statement
+  | else_if_statement
+  | else_statement
+  | empty_statement
+  | executable_statement
+  | finally_statement
+  | for_or_for_each_statement
+  | if_statement
+  | loop_statement
+  | next_statement
+  | select_statement
+  | sync_lock_statement
+  | try_statement
+  | using_statement
+  | while_statement
+  | with_statement
+  ;
+
+case_statement
+  : 'Case' (case_clause(',' case_clause)*)?
+  ;
+
+case_clause
+  : else_case_clause
+  | range_case_clause
+  | relational_case_clause
+  | simple_case_clause
+  ;
+
+else_case_clause
+  : 'Else'
+  ;
+
+range_case_clause
+  : expression 'To' expression
+  ;
+
+relational_case_clause
+  : 'Is'? ('=' | '<>' | '<' | '<=' | '>=' | '>') expression
+  ;
+
+simple_case_clause
+  : expression
+  ;
+
+catch_statement
+  : 'Catch' identifier_name? simple_as_clause? catch_filter_clause?
+  ;
+
+simple_as_clause
+  : 'As' attribute_list* type
+  ;
+
+catch_filter_clause
+  : 'When' expression
+  ;
+
+declaration_statement
+  : attributes_statement
+  | end_block_statement
+  | enum_block
+  | enum_member_declaration
+  | enum_statement
+  | event_block
+  | field_declaration
+  | imports_statement
+  | incomplete_member
+  | inherits_or_implements_statement
+  | method_base
+  | method_block_base
+  | namespace_block
+  | namespace_statement
+  | option_statement
+  | property_block
+  | type_block
+  | type_statement
+  ;
+
+end_block_statement
+  : 'End' ('If' | 'Using' | 'With' | 'Select' | 'Structure' | 'Enum' | 'Interface' | 'Class' | 'Module' | 'Namespace' | 'Sub' | 'Function' | 'Get' | 'Set' | 'Property' | 'Operator' | 'Event' | 'AddHandler' | 'RemoveHandler' | 'RaiseEvent' | 'While' | 'Try' | 'SyncLock')
+  ;
+
+enum_block
+  : enum_statement statement* end_block_statement
```

Comment:
not great.   this should be something more like `enum_statement statement* end_enum_statement`.

Reply:
investigated.  unfortunately, this really is how syntax.xml is actually structured.  it would be nice to improve on this.

Reply:
Have improved this.